### ISSUE TITLE
BN uses wrong enumeration

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#3738]
- Binary Ninja Version: 3.2.3906-dev
- OS: Windows 11, x64

**Bug Description:**
I've a enumeration like this:

```c
enum _err_t : uint32_t
{
    NONE = 0x0,
    OK = 0x1,
    FAILED = 0xffffffff,
};
```

And there is a assignation of `FAILED` to a variable. But BN uses `~NONE` instead of `FAILED`.
 
**Steps To Reproduce:**
I don't know how to reproduce this. But I can share the BNDB via PM in slack.

**Expected Behavior:**
I expect it to show `FAILED` instead of `~NONE`.

**Screenshots:**
![enum_fail](https://user-images.githubusercontent.com/109607896/207847541-cbeaf9fc-7908-4adb-a35c-f77f6ea2bbec.png)

**Comment by negasora** [comment]
I'm not able to reproduce this with a simple case on `3.2.3983-dev`.

If this still happens, can you send me the bndb in slack?

**Comment by negasora** [comment]
Confirmed still happens with the following repro steps:
1. Create a new binary view
2. Paste `85c0b9ffffffffba010000000f44ca8bc1c3` from raw hex
3. Create x86_64 function at 0
4. Define `_err_t` as above
5. In HLIL, set the type of `rcx` to `enum _err_t`
6. Bug happens

**Comment by negasora** [comment]
Fixed in 3.3.4001-dev
